src/map/parser.ts

```diff
@@ -50,7 +50,7 @@ export class MapParser {
 	}
 
 	public parseStrongKey(key: Strong, value: unknown, _parseState: State): void {
-		if (!key || !value) {
+		if (!key || value == null) {
```

Comment:
When parsing keys the `null` check might not be what we want here. Generally it is valid for values to be `null`, but we bail out on `undefined`. I may be missing the context on this also, but I think it may need to be reversed. Drop out if `value === undefined` and continue if its `null`.